cli/ts/commands/publish.ts

```diff
@@ -129,3 +136,108 @@ export const publish = async ({
   // we want the user to have the ephemeral private key
   return encKeypair.privKey.serialize();
 };
+
+/**
+ * Batch publish new messages to a MACI Poll contract
+ * @param {IPublishBatchArgs} args - The arguments for the publish command
+ * @returns {IPublishBatchData} The ephemeral private key used to encrypt the message, transaction hash
+ */
+export const publishBatch = async ({
+  messages,
+  pollId,
+  maciContractAddress,
+  publicKey,
+  privateKey,
+  signer,
+  quiet = true,
+}: IPublishBatchArgs): Promise<IPublishBatchData> => {
+  banner(quiet);
+
+  if (!PubKey.isValidSerializedPubKey(publicKey)) {
+    throw new Error("invalid MACI public key");
+  }
+
+  if (!PrivKey.isValidSerializedPrivKey(privateKey)) {
+    throw new Error("invalid MACI private key");
+  }
+
+  if (pollId < 0n) {
+    throw new Error(`invalid poll id ${pollId}`);
+  }
+
+  const userMaciPubKey = PubKey.deserialize(publicKey);
+  const userMaciPrivKey = PrivKey.deserialize(privateKey);
+  const maciContract = MACIFactory.connect(maciContractAddress, signer);
+  const pollAddress = await maciContract.getPoll(pollId);
+
+  const pollContract = PollFactory.connect(pollAddress, signer);
+
+  const [maxValues, coordinatorPubKeyResult] = await Promise.all([
+    pollContract.maxValues(),
+    pollContract.coordinatorPubKey(),
+  ]);
+  const maxVoteOptions = Number(maxValues.maxVoteOptions);
+
+  // validate the vote options index against the max leaf index on-chain
+  messages.forEach(({ stateIndex, voteOptionIndex, salt, nonce }) => {
+    if (voteOptionIndex < 0 || maxVoteOptions < voteOptionIndex) {
+      throw new Error("invalid vote option index");
+    }
+
+    // check < 1 cause index zero is a blank state leaf
+    if (stateIndex < 1) {
+      throw new Error("invalid state index");
+    }
+
+    if (nonce < 0) {
+      throw new Error("invalid nonce");
+    }
+
+    if (salt && !validateSalt(salt)) {
+      throw new Error("invalid salt");
+    }
+  });
+
+  const coordinatorPubKey = new PubKey([
+    BigInt(coordinatorPubKeyResult.x.toString()),
+    BigInt(coordinatorPubKeyResult.y.toString()),
+  ]);
+
+  const encryptionKeypair = new Keypair();
```

Comment:
I see the point of using the same key here for all messages in the batch, we just assume that all messages in the batch are from the same user either way for simplicity sake (though that doesn't necessarily have to be the case), so I would suggest we move line 226 out of the loop to avoid computing the ECDH key every time; and also we should probably return this key like we do in publishMessage.